src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -213,13 +213,13 @@ private static Bundle formatProperties(Properties properties) {
   }
 
   public static String makeKey(String key) {
-    if (key.contains(".")) {
-      key = key.trim().replace(".", "_");
-    } else if (key.contains("-")) {
-      key = key.trim().replace("-", "_");
-    } else {
-      key = key.trim().replaceAll(" ", "_");
+    String[] forbiddenChars = {".", "-", " "};
+    for (String forbidden : forbiddenChars) {
+      if (key.contains(forbidden)) {
+        key = key.trim().replace(forbidden, "_");
+      }
     }
+
```

Comment:
i like this change :). Makes the code more concise. Does it do anything different compared to the old functionality? i cant seem to spot the difference

Reply:
nvm found the diff. good spot on using `replace` instead of `replaceAll`. very subtle bug 😅

Reply:
The `replaceall` confused me too. From what I understand, the main difference is supplying a regex pattern which does not seem necessary for the simple transformations we're doing.